### PR TITLE
[#13] Improve documentation and description

### LINKED ISSUE #13: Write proper `.cabal` file description
Just not to forget about this later.

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #13 

I've updated `.cabal` description. Haven't checked it yet, just took `vec` package as an example:

* https://github.com/phadej/vec/blob/6eaa4f5ee258a709cab2342233f4539e73fd393a/vec/vec.cabal#L8-L60

Also I've update GHC versions in `.cabal` file and on Travis CI. And performed a couple more documentation improvements.

### CODE REVIEW

File: co-log-core/co-log-core.cabal [hunk 1]
```diff
@@ -1,7 +1,20 @@
+cabal-version:       2.0
 name:                co-log-core
 version:             0.1.1
-description:         Logging library
-synopsis:            Logging library
+synopsis:            Composable Contravariant Comonadic Logging Library
+description:
+    This package provides core types and function to work with @LogAction@ data type which is both simple and powerful.
```
Comment (vrom911) [inline co-log-core/co-log-core.cabal]:
```suggestion
    This package provides core types and functions to work with the @LogAction@ data type which is both simple and powerful.
```

File: co-log-core/CHANGELOG.md [hunk 2]
```diff
@@ -4,19 +4,20 @@ Change log
 `co-log-core` uses [PVP Versioning][1].
 The change log is available [on GitHub][2].
 
-0.1.1
+Unreleased: 0.1.1
```
Comment (cronokirby) [inline co-log-core/CHANGELOG.md]: I like having this tag in the CHANGELOG. Can't count the number of times I lost track of whether or not we had already released the current working version or not :laughing:

File: co-log-core/src/Colog/Core/IO.hs [hunk 3]
```diff
@@ -65,6 +60,43 @@ foo
...
+
+{- | Action that prints to stderr using 'Show'.
+
+>>> unLogAction logPrintStderr 5
+5
+-}
+logPrintStderr :: forall a m . (Show a, MonadIO m) => LogAction m a
+logPrintStderr = logPrintHandle stderr
+
+{- | Action that prints to 'Handle' using 'Show'.
```
Comment (cronokirby) [inline co-log-core/src/Colog/Core/IO.hs]: Very minor, but looking at the way `Handle` is described in [other docs](https://hackage.haskell.org/package/base-4.8.2.0/docs/GHC-IO-Handle.html), we should probably do the following:
```suggestion
{- | Action that prints to a 'Handle' using 'Show'.
```

File: co-log-core/src/Colog/Core/IO.hs [hunk 4]
```diff
@@ -65,6 +60,43 @@ foo
...
+
+>>> unLogAction (logPrintHandle stderr) 5
+5
+-}
+logPrintHandle :: forall a m . (Show a, MonadIO m) => Handle -> LogAction m a
+logPrintHandle handle = LogAction $ liftIO . hPrint handle
+
+{- | Action that prints to file using 'Show'. See 'withLogStringFile' for details.
```
Comment (cronokirby) [inline co-log-core/src/Colog/Core/IO.hs]: Same as the previous comment about `Handle`s :)
```suggestion
{- | Action that prints to a file using 'Show'. See 'withLogStringFile' for details.
```

**Comment by ChShersh** [comment]
@sphaso Well, we already kind have `example` folder:

* https://github.com/kowainik/co-log/tree/master/co-log/example

But maybe in future we can write more examples with documentation and have different executables.

**Comment by ChShersh** [comment]
@vrom911 @cronokirby @sphaso Please, re-review, this is a pre-release PR now.

**Pull request merged**